packages/flame/lib/src/components/input/button_component.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:meta/meta.dart';
+
+import '../../../components.dart';
+import '../../../extensions.dart';
+import '../../../input.dart';
+import '../../gestures/events.dart';
+
+/// The [ButtonComponent] bundles two [PositionComponent]s, one that shows while
+/// the button is being pressed, and one that shows otherwise.
+///
+/// Note: You have to set the [button] in [onLoad] if you are not passing it in
+/// through the constructor.
+class ButtonComponent extends PositionComponent with Tappable {
+  late final PositionComponent? button;
```

Comment:
can this be non-nullable?

Reply:
No, because then we can't do an assertion on `onMount`.

Reply:
Do we need an assertion though? If a variable is declared `late final` and you use it before it is initialized, Dart will through an assertion error for you automatically.

Reply:
Those runtime errors are hard for the user to understand though, but maybe they are enough.
If we decide those are enough, then we need to change on all other `PositionComponent`s that can be initialized like this too, so it is outside of the scope of this PR since it only follows the current structure.